File: java/src/main/java/org/urlshortener/dal/entities/ShortUrl.java
package org.urlshortener.dal.entities;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.validator.constraints.URL;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Size;

@Getter
@Setter
@Entity
@Table(uniqueConstraints = {@UniqueConstraint(columnNames = {"shortName"}, name = Constraints.shortUrlConstraint)})
@NoArgsConstructor
@AllArgsConstructor
public class ShortUrl {
    @Id
    @GeneratedValue
    private Long id;
    @NotBlank
    @URL
    private String url;
    @Size(min = 2, max = 10)
    private String shortName;
    @ManyToOne
    private User user;
}

File: src/main/kotlin/com/urlshortener/dal/entities/ShortUrl.kt
package com.urlshortener.dal.entities

import org.hibernate.validator.constraints.URL
import javax.persistence.*
import javax.validation.constraints.NotBlank
import javax.validation.constraints.Size

@Entity
@Table(uniqueConstraints = [UniqueConstraint(columnNames = ["shortName"], name = "ShortNameUniqueConstraint")])
data class ShortUrl(
    @field:NotBlank
    @field:URL
    var url: String,
    @field:Size(min = 2, max = 10)
    var shortName: String,
    var userId: String,
    @Id @GeneratedValue var id: Long? = null
)
